gluten-core/src/main/scala/io/glutenproject/execution/BasicPhysicalOperatorTransformer.scala

```diff
@@ -540,4 +553,36 @@ object FilterHandler {
       case other =>
         throw new UnsupportedOperationException(s"${other.getClass.toString} is not supported.")
     }
+
+  def filterExpressions(filters: Seq[Expression]): Seq[Expression] = {
+    val resFilters = filters
+      .filter {
+        case Equality(ar: Attribute, l: Literal) => checkAttribute(ar)
```

Comment:
Thanks for your fix! I think it may be better to do the check on native side. Right? cc @rui-mo.

Reply:
@PHILO-HE Could you please explain why check on native side is better? this part belongs to plan conversion, so theoretically it should be the job of java side.

Reply:
We prefer the native validation for better code maintenance. Rui may have discussed with you. Thanks!